demo/js/app.js

```diff
@@ -57,6 +58,12 @@ function Chart() {
     };
 }
 
+function RestController ($scope, RESTFactory) {
+
+    $scope.factoryBooks = RESTFactory.readObject('data/books.json');
```

Comment:
readObject is for JSON object, please use readList for JSON array which seems to be books.json.